src/ConnectionPool.jl

```diff
@@ -237,12 +272,21 @@ end
 Wait for prior pending writes to complete.
 """
 function IOExtras.startwrite(t::Transaction)
-    @require !iswritable(t)                     ;t.c.writecount != t.sequence &&
-                                                   @debug 1 "⏳  Wait write: $t"
-    while t.c.writecount != t.sequence
-        wait(t.c.writedone)
-    end                                           ;@debug 2 "👁  Start write:$t"
-    t.c.writebusy = true
+    @require !iswritable(t)
```

Comment:
Why do (and did) we appear to assert that there wasn't a previous write, then immediate wait for a previous write to finish?? This could use some comments.

Reply:
`@require !iswritable(t)` just asserts that _this specific_ transaction hasn't been written yet, so we require that first, and then try to acquire the write lock of the connection, if we get it, but the ordering doesnt' match yet (i.e. we need to wait for a previously pipelined request to finish first), that's when we wait for another `closewrite` to notify that it's done and we'll check if it's our turn.